src/wallpaper.c

```diff
@@ -245,6 +245,98 @@ static void feh_wm_set_bg_maxed(Pixmap pmap, Imlib_Image im, int use_filelist,
 	return;
 }
 
+/*
+** Creates a script that can be used to create the same background
+** as the last time the program was called.
+*/
+void feh_wm_gen_bg_script(char* fil, int centered, int scaled, int filled, int use_filelist) {
+	char * home = getenv("HOME");
+
+	if (!home)
+		return;
+
+	FILE *fp;
+	int fd;
+	char *path;
+	char *exec_absolute_path;
+	char *absolute_path;
+	struct stat s;
+	gib_list *filelist_pos = filelist;
+	exec_absolute_path = feh_absolute_path(cmdargv[0]);
```

Comment:
This only works when running a custom feh build using `./feh`, `src/feh` or similar. `feh_absolute_path` assumes that its argument (if it's not already an absolute path) is relative to the current working directory, so it doesn't work as intended when feh is in $PATH.

In the majority of cases, cmdargv[0] will be just "feh", so .fehbg will contain $pwd/feh – which most likely does not exist.

Reply:
The code I just pushed should apply the change you suggested. It checks using strchr and then sets the renamed exec_absolute_path, now exec_method, to either the converted path or as raw cmdargv[0]. The memory free also has been updated to ensure cmdargv[0] isn't free'd

Reply:
Yup, that looks good. Thanks for the patch!